src/templates/cmp.js

```diff
@@ -189,4 +189,42 @@
   } else {
     loadCmpApi(3);
   }
+
+  // send device ids
+  function sendDeviceId(consent, retriesLeft) {
+    if (retriesLeft < 0) {
+      return;
+    }
+
+    if (!__hbb_tracking_tgt || !__hbb_tracking_tgt.getDID) {
+      setTimeout(function () {
+        sendDeviceId(consent, retriesLeft - 1);
+      }, 100);
+      return;
+    }
+
+    __hbb_tracking_tgt.getDID(function (deviceId) {
+      var image = document.createElement('img');
+      image.src =
+        '<%-SUBMIT_CONSENT_FOR_TRACKING_DEVICE_ID_URL%>/' +
+        deviceId +
+        '/' +
+        Date.now() +
+        '/consent.gif?consent=' +
+        consent;
+    });
+  }
+
+  window.__tcfapi('onLogEvent', 2, function (log) {
+    var consent = undefined;
+    if (log.success === true && (log.event === 'getTCData' || log.event === 'setConsent')) {
+      consent = log.parameters.consent;
+    }
+
+    if (consent !== undefined) {
+      try {
+        sendDeviceId(consent, 3);
```

Comment:
to increase readability I'd wrap this in a function, e.g. `waitForTrackingScriptAndSendDeviceId` which internally calls `sendDeviceId(consent, 3)`